4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,81 @@
+﻿using AnimalShelterCode;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AnimalShelterCode
+{
+    public class AnimalShelter
+    {
+        public int AnimalCounter { get; private set; }
+
+        public List<Animal> Animals { get; private set; }
+
+        public AnimalShelter()
+        {
+            Animals = new List<Animal>();
+            AnimalCounter = 1;
+        }
+
+        public OperationResult AddAnimal(Animal animal)
+        {
+            if (animal.Type == AnimalType.Bear || animal.Type == AnimalType.Deer) 
+            { 
+                return new OperationResult($"Animal Type {animal.Type} is not Supported");
+            }
+            else
+            {
+                animal.UniqueAnimalId = AnimalCounter++;
```

Comment:
Though in this very specific example this will work, it is dangerous to use int's for a "unique id". This can break very easily by the following:

- Multi-tenant system(s)
- Multi-threaded system(s)
- Restarting the application (as memory is cleaned and started a new)
- Restarting the operating system
- Saving the animal and or loading the animal from a disk / database

The list goes on and on. For unique id's I would suggest using the Guid type for it can create a new globally unique id (unique in the sense that even if every single computer was to attempt to generate the same one it would take thousands of years to get a duplicate - and its a low percentage that would happen).